central/deployment/datastore/datastore_impl_postgres_test.go

```diff
@@ -0,0 +1,343 @@
+
+	mapping.RegisterCategoryToTable(v1.SearchCategory_IMAGE_COMPONENTS, schema.ImageComponentsSchema)
+	mapping.RegisterCategoryToTable(v1.SearchCategory_IMAGE_VULNERABILITIES, schema.ImageCvesSchema)
+	mapping.RegisterCategoryToTable(v1.SearchCategory_IMAGE_COMPONENT_EDGE, schema.ImageComponentEdgesSchema)
+	mapping.RegisterCategoryToTable(v1.SearchCategory_COMPONENT_VULN_EDGE, schema.ImageComponentCveEdgesSchema)
+	mapping.RegisterCategoryToTable(v1.SearchCategory_IMAGE_VULN_EDGE, schema.ImageCveEdgesSchema)
+}
+
+func (s *DeploymentPostgresDataStoreTestSuite) TearDownSuite() {
+	mapping.UnregisterCategory(v1.SearchCategory_IMAGE_COMPONENTS)
+	mapping.UnregisterCategory(v1.SearchCategory_IMAGE_VULNERABILITIES)
+	mapping.UnregisterCategory(v1.SearchCategory_IMAGE_COMPONENT_EDGE)
+	mapping.UnregisterCategory(v1.SearchCategory_COMPONENT_VULN_EDGE)
+	mapping.UnregisterCategory(v1.SearchCategory_IMAGE_VULN_EDGE)
+
+	s.db.Close()
+	pgtest.CloseGormDB(s.T(), s.gormDB)
+	s.mockCtrl.Finish()
+	s.envIsolator.RestoreAll()
+}
+
+func (s *DeploymentPostgresDataStoreTestSuite) TestSearchWithPostgres() {
+	ctx := sac.WithAllAccess(context.Background())
+	img1 := fixtures.GetImageWithUniqueComponents()
+	img1.Id = uuid.NewV4().String()
+	img2 := fixtures.GetImageWithUniqueComponents()
+	img2.Id = uuid.NewV4().String()
+	img2.Scan.OperatingSystem = "pluto"
+	for _, component := range img2.GetScan().GetComponents() {
+		component.Name = img2.Id + component.Name
+		for _, vuln := range component.GetVulns() {
+			vuln.Cve = img2.Id + vuln.Cve
+		}
+	}
+	img3 := fixtures.GetImageWithUniqueComponents()
+	img3.Id = uuid.NewV4().String()
+
+	dep1 := fixtures.GetDeploymentWithImage("c1", "n1", img1)
+	dep2 := fixtures.GetDeploymentWithImage("c1", "n2", img2)
+	dep3 := fixtures.GetDeploymentWithImage("c2", "n1", img3)
+
+	// Upsert images.
+	s.NoError(s.imageDatastore.UpsertImage(ctx, img1))
+	s.NoError(s.imageDatastore.UpsertImage(ctx, img2))
+	s.NoError(s.imageDatastore.UpsertImage(ctx, img3))
+	// Upsert Deployments.
+	s.NoError(s.deploymentDatastore.UpsertDeployment(ctx, dep1))
+	s.NoError(s.deploymentDatastore.UpsertDeployment(ctx, dep2))
+	s.NoError(s.deploymentDatastore.UpsertDeployment(ctx, dep3))
+
+	for _, tc := range []struct {
+		desc         string
+		ctx          context.Context
+		query        *v1.Query
+		orderMatters bool
+		expectedIDs  []string
+		queryImages  bool
+	}{
+		{
+			desc:         "Search deployments with empty query",
+			ctx:          ctx,
+			query:        pkgSearch.EmptyQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{dep1.Id, dep2.Id, dep3.Id},
+		},
+		{
+			desc:         "Search deployments with query",
+			ctx:          ctx,
+			query:        pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.DeploymentID, dep1.Id).ProtoQuery(),
+			orderMatters: false,
+			expectedIDs:  []string{dep1.Id},
+		},
+		{
+			desc:         "Search deployments with image query",
```

Comment:
also add: search deployments with image that doesn't exist in any deployments that resolves to no results?

Basically negative search cases